From fade891a17967428f4ce5a8b672f16c878dc4805 Mon Sep 17 00:00:00 2001
From: Chris Burr <chrisburr@users.noreply.github.com>
Date: Wed, 18 Sep 2019 10:26:05 +0200
Subject: [PATCH] Fix linking for libXrdCmsRedirectLocal-4.so

---
 src/XrdPlugins.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/XrdPlugins.cmake b/src/XrdPlugins.cmake
index b605e70da02..84273b3ba0f 100644
--- a/src/XrdPlugins.cmake
+++ b/src/XrdPlugins.cmake
@@ -157,7 +157,9 @@ add_library(
   XrdCms/XrdCmsRedirLocal.cc XrdCms/XrdCmsRedirLocal.hh )
 
 target_link_libraries(
-  ${LIB_XRD_CMSREDIRL} )
+  ${LIB_XRD_CMSREDIRL}
+  XrdServer
+  XrdUtils )
 
 set_target_properties(
   ${LIB_XRD_CMSREDIRL}
